### DISCUSSION TITLE
getting Error 1: Incorrect Function.

### DISCUSSION THREAD

**Discussion by Munib01** [strongswan/strongswan#1741] [Q&A]
I have followed strongswan documentation, [link](https://docs.strongswan.org/docs/5.9/os/windows.html), to compile an executable for windows. 
I have successfully created the executable. I have created the service using the following command :
```
sc create "strongSwan IKE service name" binPath= path_to_exe
```
However, when I try to run the service through terminal, I get following log:
```
SERVICE_NAME: strongSwan IKE service name
        TYPE               : 10  WIN32_OWN_PROCESS
        STATE              : 2  START_PENDING
                                (NOT_STOPPABLE, NOT_PAUSABLE, IGNORES_SHUTDOWN)
        WIN32_EXIT_CODE    : 0  (0x0)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x7d0
        PID                : 17284
        FLAGS              :

```

And, If I try to run it through the Service App available in windows, I get following error ("strongSwan IKE service 9" is the name of the service):
![image](https://github.com/strongswan/strongswan/assets/53575827/087967ba-d168-4423-a0c3-7fdb3ac6e2d8)

I have gone through the internet but could not Understand what is the problem. Can anyone guide me anything about this????

**Comment by tobiasbrunner** [comment]
No idea. Any information in the event log? Can you start `charon-svc.exe` manually in the console?

**Reply by Munib01** [reply]
I have seen ```Event viewer``` but could not find any log specific to strongswan

however, These are the logs I get while running ```charon-svc.exe``` in the console:
```
C:\Users\Emumba\Downloads\strongswan>C:\Users\Emumba\Downloads\strongswan\src\charon-svc\charon-svc.exe
00[DMN] Starting IKE service charon-svc (strongSwan 5.9.10, Windows Client 10.0.19045 (SP 0.0)
00[LIB] providers loaded by OpenSSL: default
00[LIB] loaded plugins: charon-svc nonce x509 pubkey pkcs1 pem openssl kernel-wfp kernel-iph socket-win vici
00[JOB] spawning 16 worker threads
```
and following is the ```strongswan.conf``` i am using:
```
# /etc/strongswan.conf - strongSwan configuration file

charon-systemd {
  load = random nonce aes sha1 sha2 md5 pem pkcs1 mgf1 gmp hmac gcm curve25519 kdf x509 revocation curl vici kernel-netlink socket-default eap-identity eap-ttls eap-md5 eap-tnc tnc-tnccs tnccs-dynamic tnccs-11 tnccs-20 tnc-imv updown

  multiple_authentication=no
  integrity_test = yes

  syslog {
    daemon {
      tnc = 3
      imv = 3
    }
  }
  plugins {
    eap-ttls {
      phase2_method = md5
      phase2_piggyback = yes
      phase2_tnc = yes
      phase2_tnc_method = tnc
    }
    eap-tnc {
      protocol = tnccs-dynamic
    }
  }
}

libtls {
  version_max = 1.3
  suites = TLS_AES_128_GCM_SHA256, TLS_DHE_RSA_WITH_AES_128_CBC_SHA256
}
```

**Reply by tobiasbrunner** [reply]
> however, These are the logs I get while running `charon-svc.exe` in the console:

Looks fine. So no idea what's going wrong when started as a service. Maybe the path is incorrect?

> and following is the `strongswan.conf` i am using:

That config is pretty much useless as you are running `charon-svc` and not `charon-systemd`, so only the settings in the `libtls` section will apply (if you use any TLS-based EAP methods, of which none are loaded). Only settings in the `charon` section are inherited by the other daemons. And note that the plugin list is obviously invalid for Windows (and the other settings also make only sense with certain plugin configurations).
